documentation/modules/ROOT/pages/create.adoc

```diff
@@ -0,0 +1,214 @@
+
+Modify the name of the values file with the image values with:
+
+[.console-input]
+[source, yaml,subs="attributes+,+macros"]
+----
+image:
+  repository: ammbra/faq
+  tag: "1.0.0"
+  pullPolicy: IfNotPresent
+  containerPort: 8080
+----
+
+Using the details from previous section when you installed the database, you can parameterize the connection to the database:
+
+[.console-input]
+[source, yaml, subs="normal,attributes"]
+----
+postgresql:
+  server: faq-db-postgresql
+  postgresqlUsername: faq-default
+  secretName: faq-db-postgresql
+  secretKey: postgres-password
+----
+
+Every well-designed application needs health checks and this why you should add the following:
+
+[.console-input]
+[source, yaml,subs="attributes+,+macros"]
+----
+readinessProbe:
+  path: /q/health/ready
+  initialDelaySeconds: 5
+  timeoutSeconds: 3
+  periodSeconds: 3
+  failureThreshold: 3
+
+
+livenessProbe:
+  path: /q/health/live
+  initialDelaySeconds: 10
+  timeoutSeconds: 2
+  periodSeconds: 8
+  failureThreshold: 3
+----
+
+Your deployed application should be accessible from inside and outside the Kubernetes cluster.
+A Kubernetes Service of type `LoadBalancer` will be used for this installation.
+
+Please change the service values that will expose your microservice:
+
+[.console-input]
+[source, yaml,subs="attributes+,+macros"]
+----
+service:
+  type: LoadBalancer
+  port: 8080
+----
+
+== Modify the template associated with Deployment
+
+Navigate to `templates/deployment.yaml`.
+The keys associated to the values defined in `values.yaml` need to be employed in corresponding templates by using template directives.
+
+NOTE: A template directive is enclosed in `{{ and }}` blocks.
+
+
+Modify the Docker image section by adding a parametrizable container port:
+
+[.console-input]
+[source, yaml,subs="attributes+,+macros"]
+----
+image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+imagePullPolicy: {{ .Values.image.pullPolicy }}
+ports:
+  - name: http 
+    containerPort: {{ .Values.image.containerPort }}
+----
+
+Since the microservice is also using a PostgreSQL database, let's add its details:
+
+[.console-input]
+[source, yaml,subs="attributes+,+macros"]
+----
+env:
+  - name: POSTGRES_SERVER
+    value: {{ .Values.postgresql.server | default (printf "%s-postgresql" ( .Release.Name )) | quote }} <1>
+  - name: POSTGRES_USERNAME
+    value: {{ .Values.postgresql.postgresqlUsername | default (printf "postgres" ) | quote }}
+  - name: POSTGRES_PASSWORD
+    valueFrom:
+      secretKeyRef:
+        name: {{ .Values.postgresql.secretName | default (printf "%s-postgresql" ( .Release.Name )) | quote }}
+        key: {{ .Values.postgresql.secretKey }} <3>
+----
+
+<1> If the value for the PostgreSQL server is defined in `values.yaml`, that one will be filled in.
+ Else, a string composed from name of the release and ending with `postgresql` will be filled in.
+
+<2> Quote the username value from `values.yaml`.
```

Comment:
Couldn't find the `<2>` in the snippet. Can you explain what is `quote`?

Reply:
Good catch! Thank you!